src/node/utils/caching_middleware.js

```diff
@@ -192,11 +191,12 @@ CachingMiddleware.prototype = new function () {
         res.write = old_res.write || res.write;
         res.end = old_res.end || res.end;
 
-        var headers = responseCache[cacheKey].headers;
+        let headers = {};
+        Object.assign(headers, (responseCache[cacheKey].headers || {}));
         var statusCode = responseCache[cacheKey].statusCode;
 
         var pathStr = CACHE_DIR + 'minified_' + cacheKey;
-        if (supportsGzip && (headers['content-type'] || '').match(/^text\//)) {
+        if (supportsGzip && /application\/javascript/.test(headers['content-type'])) {
```

Comment:
Maybe this should just be:

```suggestion
        if (supportsGzip && headers['content-type'] === 'application/javascript') {
```

Reply:
That was my second thought, too, but content-type can/should contain `charset`, too.

Reply:
Oh right, good point.